test/gas/InterchainGasEstimation.js

```diff
@@ -0,0 +1,36 @@
+'use strict';
+
+const chai = require('chai');
+const { expect } = chai;
+const { ethers } = require('hardhat');
+
+describe('InterchainGasEstimation', () => {
+    let gasEstimateFactory;
+    let gasEstimate;
+
+    let ownerWallet;
+
+    const sourceChain = 'ethereum';
+    const destinationChain = 'optimism';
+
+    before(async () => {
+        [ownerWallet] = await ethers.getSigners();
+
+        gasEstimateFactory = await ethers.getContractFactory('TestInterchainGasEstimation', ownerWallet);
+        gasEstimate = await gasEstimateFactory.deploy().then((d) => d.deployed());
+
+        await gasEstimate.updateGasInfo(sourceChain, [0, 0, 50000000000, 1]);
+        await gasEstimate.updateGasInfo(destinationChain, [1, 0, 5000, 0]);
```

Comment:
set it within the test itself? and only destination chain gas info needs to be set

Reply:
Destination as L2 + Ethereum as L1